Source/controls/touch/gamepad.cpp

```diff
@@ -62,7 +62,7 @@ void InitializeVirtualGamepad()
 	if (SDL_GetDisplayDPI(displayIndex, nullptr, &hdpi, &vdpi) == 0) {
 		int clientWidth;
 		int clientHeight;
-		SDL_GetWindowSize(ghMainWnd, &clientWidth, &clientHeight);
+		SDL_GetRendererOutputSize(renderer, &clientWidth, &clientHeight);

```

Comment:
`renderer` could be null if `Upscale=0`.

```suggestion
		if (renderer != nullptr)
			SDL_GetRendererOutputSize(renderer, &clientWidth, &clientHeight);
		else
			SDL_GetWindowSize(ghMainWnd, &clientWidth, &clientHeight);

```

Reply:
Oh, I thought the renderer was was required for drawing the touch UI :)

Reply:
Nah, I was going require upscaling since touch controls don't work on SDL1 anyway. But when you asked me a question about making the virtual gamepad depend on a tangentially related setting, I gave it a bit more thought and discovered a relatively easy way to make it work with `SDL_BlitScaled()`. It looks awful, of course, but it functions.